Improve HTTP JSONPath data adapter error message content

In a customer deployment, it was noted that an HTTP JSONPath data adapter was failing with the following message:

`2023-12-11 20:06:40,925 ERROR: org.graylog2.lookup.adapters.HTTPJSONPathDataAdapter - HTTP request error for key <1.1.1.1>
javax.net.ssl.SSLHandshakeException: None of the TrustManagers trust this certificate chain.
	at sun.security.ssl.Alert.createSSLException(Alert.java:131) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:378) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:321) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:316) ~[?:?]
	at sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:654) ~[?:?]
	at sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(CertificateMessage.java:473) ~[?:?]
	at sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(CertificateMessage.java:369) ~[?:?]
	at sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:396) ~[?:?]
	at sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:480) ~[?:?]
	at sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:458) ~[?:?]
	at sun.security.ssl.TransportContext.dispatch(TransportContext.java:201) ~[?:?]
	at sun.security.ssl.SSLTransport.decode(SSLTransport.java:172) ~[?:?]
	at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1510) ~[?:?]
	at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1425) ~[?:?]
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:455) ~[?:?]
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:426) ~[?:?]
	at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.kt:379) ~[graylog.jar:?]
	at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.kt:337) ~[graylog.jar:?]
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:209) ~[graylog.jar:?]
	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226) ~[graylog.jar:?]
	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106) ~[graylog.jar:?]
	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74) ~[graylog.jar:?]
	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255) ~[graylog.jar:?]
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109) ~[graylog.jar:?]
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109) ~[graylog.jar:?]
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109) ~[graylog.jar:?]
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109) ~[graylog.jar:?]
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201) ~[graylog.jar:?]
	at okhttp3.internal.connection.RealCall.execute(RealCall.kt:154) ~[graylog.jar:?]
	at org.graylog2.lookup.adapters.HTTPJSONPathDataAdapter.doGet(HTTPJSONPathDataAdapter.java:188) ~[graylog.jar:?]
	at org.graylog2.plugin.lookup.LookupDataAdapter.get(LookupDataAdapter.java:143) ~[graylog.jar:?]
	at org.graylog2.lookup.LookupTable.lambda$lookup$0(LookupTable.java:73) ~[graylog.jar:?]
	at org.graylog2.lookup.caches.CaffeineLookupCache.lambda$get$1(CaffeineLookupCache.java:131) ~[graylog.jar:?]
	at com.github.benmanes.caffeine.cache.LocalCache.lambda$statsAware$0(LocalCache.java:139) ~[graylog.jar:?]
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2380) ~[graylog.jar:?]
	at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1955) [?:?]
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2378) [graylog.jar:?]
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2361) [graylog.jar:?]
	at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:108) [graylog.jar:?]
	at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:62) [graylog.jar:?]
	at org.graylog2.lookup.caches.CaffeineLookupCache.get(CaffeineLookupCache.java:146) [graylog.jar:?]
	at org.graylog2.lookup.LookupTable.lookup(LookupTable.java:73) [graylog.jar:?]
	at org.graylog2.lookup.LookupTableService$Function.lookup(LookupTableService.java:633) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.functions.lookup.LookupValue.evaluate(LookupValue.java:66) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.ast.expressions.FunctionExpression.evaluateUnsafe(FunctionExpression.java:63) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.ast.expressions.Expression.evaluate(Expression.java:36) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.ast.statements.VarAssignStatement.evaluate(VarAssignStatement.java:33) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.ast.statements.VarAssignStatement.evaluate(VarAssignStatement.java:22) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.processors.PipelineInterpreter.evaluateStatement(PipelineInterpreter.java:399) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.processors.PipelineInterpreter.executeRuleActions(PipelineInterpreter.java:383) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.processors.PipelineInterpreter.evaluateStage(PipelineInterpreter.java:335) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.processors.PipelineInterpreter.processForResolvedPipelines(PipelineInterpreter.java:279) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.processors.PipelineInterpreter.process(PipelineInterpreter.java:151) [graylog.jar:?]
	at org.graylog.plugins.pipelineprocessor.processors.PipelineInterpreter.process(PipelineInterpreter.java:107) [graylog.jar:?]
	at org.graylog2.shared.buffers.processors.ProcessBufferProcessor.handleMessage(ProcessBufferProcessor.java:161) [graylog.jar:?]
	at org.graylog2.shared.buffers.processors.ProcessBufferProcessor.dispatchMessage(ProcessBufferProcessor.java:131) [graylog.jar:?]
	at org.graylog2.shared.buffers.processors.ProcessBufferProcessor.onEvent(ProcessBufferProcessor.java:101) [graylog.jar:?]
	at org.graylog2.shared.buffers.processors.ProcessBufferProcessor.onEvent(ProcessBufferProcessor.java:49) [graylog.jar:?]
	at com.lmax.disruptor.WorkProcessor.run(WorkProcessor.java:143) [graylog.jar:?]
	at com.codahale.metrics.InstrumentedThreadFactory$InstrumentedRunnable.run(InstrumentedThreadFactory.java:66) [graylog.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.security.cert.CertificateException: None of the TrustManagers trust this certificate chain.
	at org.graylog2.security.CustomCAX509TrustManager.checkServerTrusted(CustomCAX509TrustManager.java:87) ~[graylog.jar:?]
	at sun.security.ssl.AbstractTrustManagerWrapper.checkServerTrusted(SSLContextImpl.java:1441) ~[?:?]
	at sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:638) ~[?:?]
	... 60 more`

The root cause was that the SSL certificate had expired for the remote endpoint that the customer had set in their JSON data adapter. This cannot be inferred from the message presented above.

The error message could be significantly improved with at least the name or ID of the data adapter where the issue was encountered, and/or the Lookup URL of the endpoint that was attempting to be reached. The customer's environment had 86 data adapters, necessitating scrolling through them in the web UI as the error message does not signify the data adapter having a problem, only that a problem existed. Additionally, a statement that the SSL certificate had expired would have been helpful too, if this is possible. 

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

* Graylog Version: Graylog Cloud 5.2.2 (397)

--- a/changelog/unreleased/issue-17649.toml
+++ b/changelog/unreleased/issue-17649.toml
@@ -1,0 +1,5 @@
+type = "c"
+message = "Provide more verbose error messages for HTTPJSONPath data adapter."
+
+issues = ["17649"]
+pulls = [""]

--- a/changelog/unreleased/issue-17649.toml
+++ b/changelog/unreleased/issue-17649.toml
@@ -2,4 +2,4 @@ type = "c"
 message = "Provide more verbose error messages for HTTPJSONPath data adapter."
 
 issues = ["17649"]
-pulls = [""]
+pulls = ["17801"]

--- a/graylog2-server/src/main/java/org/graylog2/lookup/adapters/HTTPJSONPathDataAdapter.java
+++ b/graylog2-server/src/main/java/org/graylog2/lookup/adapters/HTTPJSONPathDataAdapter.java
@@ -160,7 +160,7 @@ public class HTTPJSONPathDataAdapter extends LookupDataAdapter {
         final String urlString = templateEngine.transform(config.url(), ImmutableMap.of("key", encodedKey));
 
         if (!urlWhitelistService.isWhitelisted(urlString)) {
-            LOG.error("URL <{}> is not whitelisted. Aborting lookup request.", urlString);
+            LOG.error("Data adapter <{}>: URL <{}> is not whitelisted. Aborting lookup request.", name(), urlString);
             publishSystemNotificationForWhitelistFailure();
             setError(UrlNotWhitelistedException.forUrl(urlString));
             return getErrorResult();
@@ -173,7 +173,7 @@ public class HTTPJSONPathDataAdapter extends LookupDataAdapter {
         final HttpUrl url = HttpUrl.parse(urlString);
 
         if (url == null) {
-            LOG.error("Couldn't parse URL <{}> - returning empty result", urlString);
+            LOG.error("Data adapter <{}>: Couldn't parse URL <{}> - returning empty result", name(), urlString);
             httpURLErrors.mark();
             return getErrorResult();
         }
@@ -198,7 +198,7 @@ public class HTTPJSONPathDataAdapter extends LookupDataAdapter {
             }
             return result;
         } catch (IOException e) {
-            LOG.error("HTTP request error for key <{}>", key, e);
+            LOG.error("Data adapter <{}>: HTTP request error for key <{}> from URL <{}>", name(), key, urlString, e);
             httpRequestErrors.mark();
             return getErrorResult();
         } finally {

